core/common/src/test/java/alluxio/ConfigurationDocGeneratorTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.List;
+import java.util.ArrayList;
+
+/**
+ * Tests for {@link ConfigurationDocGenerator}.
+ */
+public class ConfigurationDocGeneratorTest {
+  private static final String FILE_HEADER = "propertyName,defaultValue";
+  private String mLocation;
+
+  /** Rule to create a new temporary folder during each test. */
+  @Rule
+  public TemporaryFolder mFolder = new TemporaryFolder();
+
+  /**
+   * Sets up all dependencies before a test runs.
+   */
+  @Before
+  public void backupCSVFiles() throws Exception {
+    mLocation = mFolder.newFolder().toString() + "/";
```

Comment:
Since  mFolder.newFolder() needs to handle IOException, I put it into backupCSVFiles.